src/main/java/sc/iview/SciView.kt

```diff
@@ -803,8 +806,9 @@ class SciView : SceneryBase, CalibratedRealInterval<CalibratedAxis> {
      * @return a Node corresponding to the Node
      */
     @JvmOverloads
-    fun addNode(n: Node?, activePublish: Boolean = true): Node? {
+    fun <N: Node?> addNode(n: N, activePublish: Boolean = true, block: N.() -> Unit = {}): N {
```

Comment:
Why is this `Node?` and not `Node`?

Reply:
I don't know - that's just how it was - I can change it to `Node`, but this would be backwards incompatible. This way you can pass anything, also `null`, but if it is not `null`, it will return the actual type and not just `Node` which is nice.

Reply:
@skalarproduktraum should I break backwards compatibility here and change it to `Node` or keep it as is?

Reply:
I think you can do that, there's I think no reason why this function should be called with null as parameter. Adding a null node just makes no sense. So breaking that is fine I'd say.

Reply:
I changed this to `Node`.